src/chol_cache.h

```diff
@@ -0,0 +1,79 @@
+#ifndef CHOL_CACHE_INCLUDED_
+#define CHOL_CACHE_INCLUDED_
+
+#include "covariance.h"
+#include "utils.h"
+// Base class of spatial and non-spatial Cholesky.
+template <class Type>
+struct lower_chol_base {
+  virtual matrix<Type> get_chol(std::vector<int> visits, matrix<Type> dist) = 0;
+};
+// Struct to obtain Cholesky for non-spatial.
+template <class Type>
+struct lower_chol_nonspatial: public lower_chol_base<Type> {
+  std::map<std::vector<int>, matrix<Type>> chols;
+  std::string cov_type;
+  int n_visits;
+  std::vector<int> full_visit;
+  int n_theta; // theta.size()
```

Comment:
please remove any commented code, e.g. here:
```suggestion
  int n_theta;
```